### DISCUSSION TITLE
Is running with the docker-compose file supported?

### DISCUSSION THREAD

**Discussion by mwarger** [amplication/amplication#1317] [Q&A] [answered]
I'm seeing the following error when trying to start the app locally with docker-compose.  I'm not sure how to adjust the version (as I thought the dockerfile should lock the version).  I have node 14 installed (but I have access to use nvm if needed).  Any help is appreciated.

> [ 6/11] RUN cd server; npm ci --loglevel=notice;:
#10 21.22 npm WARN EBADENGINE Unsupported engine {
#10 21.22 npm WARN EBADENGINE   package: '@angular-devkit/core@11.2.7',
#10 21.22 npm WARN EBADENGINE   required: { node: '>= 10.13.0', npm: '^6.11.0 || ^7.5.6', yarn: '>= 1.13.0' },
#10 21.22 npm WARN EBADENGINE   current: { node: 'v14.15.3', npm: '7.3.0' }
#10 21.22 npm WARN EBADENGINE }
#10 21.22 npm WARN EBADENGINE Unsupported engine {
#10 21.22 npm WARN EBADENGINE   package: '@angular-devkit/schematics@11.2.7',
#10 21.22 npm WARN EBADENGINE   required: { node: '>= 10.13.0', npm: '^6.11.0 || ^7.5.6', yarn: '>= 1.13.0' },
#10 21.22 npm WARN EBADENGINE   current: { node: 'v14.15.3', npm: '7.3.0' }
#10 21.22 npm WARN EBADENGINE }

**Comment by mwarger** [comment, marked as answer]
I was able to solve the docker build issue by updating the `npm` installation line in the Dockerfile to be 
`RUN npm i -g npm@7.5.6`

**Comment by yuval-hazaz** [comment]
Interesting
we will look into it to see if wee need to change the code
Thanks for the update

**Reply by mwarger** [reply]
Would you be able to point me to where this is in the repo? It would help get my bearings on how this is setup. I'm open to PRing this change.

**Reply by yuval-hazaz** [reply]
sure, the source is here
https://github.com/amplication/amplication/blob/master/packages/amplication-data-service-generator/src/static/Dockerfile

you can change the npm version to a newer version like you did manually but it requires some test to make sure everything is still working. 
when you send a PR it will run an e2e test including docker build.
